### ISSUE TITLE
can not serialize 'TemplateResponse' object

### ISSUE THREAD

**Issue by engineervix** [Adrian-Turjak/wagtail-robots#7]
I'm using [django-redis](https://github.com/jazzband/django-redis) with serialization using *MsgPack* as described [here](http://jazzband.github.io/django-redis/latest/#_pluggable_serializer). I'm also using *Zlib Compression* as described [here](http://jazzband.github.io/django-redis/latest/#_compression_support). My settings are as follows:

```python
CACHES = {
    "default": {
        "BACKEND": "django_redis.cache.RedisCache",
        "LOCATION": "redis://127.0.0.1:6379/1",
        "KEY_PREFIX": "key_prefix_here",
        "OPTIONS": {
            "CLIENT_CLASS": "django_redis.client.DefaultClient",
            "SERIALIZER": "django_redis.serializers.msgpack.MSGPackSerializer",
            "COMPRESSOR": "django_redis.compressors.zlib.ZlibCompressor",
        },
    }
}

SESSION_ENGINE = "django.contrib.sessions.backends.cache"
SESSION_CACHE_ALIAS = "default"
```

When I [enable caching](https://wagtail-robots.readthedocs.io/en/latest/#caching) of `robots.txt`, an exception is raised when `/robots.txt` is accessed, as follows:

```
TypeError: can not serialize 'TemplateResponse' object
File "django/core/handlers/exception.py", line 34, in inner
  response = get_response(request)
File "django/core/handlers/base.py", line 145, in _get_response
  response = self.process_exception_by_middleware(e, request)
File "django/core/handlers/base.py", line 143, in _get_response
  response = response.render()
File "django/template/response.py", line 108, in render
  newretval = post_callback(retval)
File "django/utils/decorators.py", line 156, in callback
  return middleware.process_response(request, response)
File "django/middleware/cache.py", line 106, in process_response
  lambda r: self.cache.set(cache_key, r, timeout)
File "django/template/response.py", line 93, in add_post_render_callback
  callback(self)
File "django/middleware/cache.py", line 106, in <lambda>
  lambda r: self.cache.set(cache_key, r, timeout)
File "django_redis/cache.py", line 36, in _decorator
  return method(self, *args, **kwargs)
File "django_redis/cache.py", line 76, in set
  return self.client.set(*args, **kwargs)
File "django_redis/client/default.py", line 127, in set
  nvalue = self.encode(value)
File "django_redis/client/default.py", line 354, in encode
  value = self._serializer.dumps(value)
File "django_redis/serializers/msgpack.py", line 8, in dumps
  return msgpack.dumps(value)
File "__init__.py", line 35, in packb
  return Packer(**kwargs).pack(o)
File "msgpack/_packer.pyx", line 286, in msgpack._cmsgpack.Packer.pack
File "msgpack/_packer.pyx", line 292, in msgpack._cmsgpack.Packer.pack
File "msgpack/_packer.pyx", line 289, in msgpack._cmsgpack.Packer.pack
File "msgpack/_packer.pyx", line 283, in msgpack._cmsgpack.Packer._pack
```

When I disable caching, the problem goes away. Any suggestions on how I could still enable caching of `robots.txt`, but with serialization disabled just for *wagtail-robots*?

**Comment by unexceptable** [comment]
I'll see if I can add config to allow you to specify a cache name. You can then use an in memory cache for robots, while keeping the rest in the default cache.

**Comment by unexceptable** [comment]
@engineervix 
Take a look at this pull request: https://github.com/Adrian-Turjak/wagtail-robots/pull/8

Test out that branch and tell me how that works for you. That should allow you to make a second cache alias with different config, and make wagtail-robots use that other one.

**Comment by unexceptable** [comment]
fixed with release 0.3.0
